addons/website/models/website_visitor.py

```diff
@@ -265,32 +285,49 @@ def _link_to_partner(self, partner, update_values=None):
             vals.update(update_values)
         self.write(vals)
 
-    def _link_to_visitor(self, target, keep_unique=True):
+    def _link_to_visitor(self, target):
         """ Link visitors to target visitors, because they are linked to the
         same identity. Purpose is mainly to propagate partner identity to sub
         records to ease database update and decide what to do with "duplicated".
-        THis method is meant to be overridden in order to implement some specific
+        This method is meant to be overridden in order to implement some specific
         behavior linked to sub records of duplicate management.
 
         :param target: main visitor, target of link process;
-        :param keep_unique: if True, find a way to make target unique;
         """
         # Link sub records of self to target partner
         if target.partner_id:
             self._link_to_partner(target.partner_id)
         # Link sub records of self to target visitor
         self.website_track_ids.write({'visitor_id': target.id})
 
-        if keep_unique:
-            self.unlink()
+        # Archive current record and set its parent visitor
+        self.partner_id = False
+        self.parent_id = target.id
+        self.active = False
 
         return target
 
-    def _cron_archive_visitors(self):
+    def _cron_clean_inactive_visitors(self):
```

Comment:
``_cron_unlink_old_visitors`` ? Using unlink to indicate we now unlink them (cleaning could be something else), and "inactive" is misleading as we also include not archived visitors.